qualtran/bloqs/arithmetic.py

```diff
@@ -578,6 +578,109 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
         return {(Add(self.bitsize), 1), (ArbitraryClifford(n=2), self.bitsize)}
 
 
+@frozen
+class HammingWeightCompute(GateWithRegisters):
+    r"""A gate to compute the hamming weight of n-bit register in a new log_{n} bit register.
+
+    Implements $U|x\rangle |0\rangle \rightarrow |x\rangle|hamming_weight(x)\rangle$
+    using $4nT$ gates. Uncomputation requires 0 T-gates.
```

Comment:
Ian's paper and the Hubbard paper both say n-1 toffoli  But Craig's paper says n Toffoli.  Is there an accounting error? or a reporting error?